pkg/client/dtclient/client_settings.go

```diff
@@ -247,6 +253,76 @@ func (d *DynatraceClient) upsertSettings(ctx context.Context, obj SettingsObject
 	return entity, nil
 }
 
+func isSameValueOfConstraint(key string, c1 []byte, c2 []byte) (bool, error) {
+	u := make(map[string]any)
+	if err := json.Unmarshal(c1, &u); err != nil {
+		return false, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+	}
+	v1 := u[key]
+
+	if err := json.Unmarshal(c2, &u); err != nil {
+		return false, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+	}
+	v2 := u[key]
+
+	return v1 == v2, nil
```

Comment:
this check does not work if the unique value is an object instead of a value. 

So e.g. if the key is "A" and it's value is another JSON object `{"key":"x", "val":"y"}` this will panic.

I've added a test case and adapt this accordingly in a commit.
I'm not sure if that case is possible, but we should not rule it out and fail execution if it happens.